File: mobileapps/android/games/TheFlyingDragon/app/src/main/java/com/example/tungnguyen/theflyingdragon/MainActivity.java
package com.example.tungnguyen.theflyingdragon;

import android.graphics.drawable.AnimationDrawable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public void onWindowFocusChanged(boolean hasFocus) {
        super.onWindowFocusChanged(hasFocus);
        if(hasFocus){
            Animation animation = AnimationUtils.loadAnimation(this, R.anim.sun_rise);
            findViewById(R.id.imageView).startAnimation(animation);
            Animation animation1 = AnimationUtils.loadAnimation(this, R.anim.clock_turn);
            findViewById(R.id.imageView3).startAnimation(animation1);
            AnimationDrawable dragonAnimationDrawable = (AnimationDrawable) findViewById(R.id.imageView4).getBackground();
            if(!dragonAnimationDrawable.isRunning()){
                dragonAnimationDrawable.start();
            }
            Animation dragonFlying = AnimationUtils.loadAnimation(this, R.anim.dragon_fly);
            findViewById(R.id.imageView4).startAnimation(dragonFlying);
        }
    }
}

File: mobileapps/android/games/Stopthebomb/app/src/main/java/com/petamind/stopthebomb/MainActivity.kt
package com.petamind.stopthebomb

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.view.View
import android.view.animation.Animation
import android.view.animation.Animation.AnimationListener
import android.view.animation.AnimationUtils
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), AnimationListener, View.OnClickListener {
    lateinit var anim: ProgressBarAnimation
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<View>(android.R.id.content).setOnClickListener(this)
        val animation = AnimationUtils.loadAnimation(this, R.anim.zoom)
        image.startAnimation(animation)
    }

    private fun reset() {
        startMusic(R.raw.grasshopper)
        image.setImageResource(R.drawable.ic_bomb)
        progress.setScaleY(3f);
        progress.progress = 0

        anim = ProgressBarAnimation(progress, 0f, 100f)
        anim.setAnimationListener(this)
        anim.duration = 2000
        button.isEnabled = true
    }

    private fun startMusic(id: Int = R.raw.maze) {
        stopMusic()
        startService(Intent(this, BackgroundSoundService::class.java).apply {
            putExtra("id", id)
        })
    }

    private fun stopMusic() {
        stopService(Intent(this, BackgroundSoundService::class.java))
    }

    fun start(v: View) {
        reset()
        startMusic()
        button.isEnabled = false
        progress.startAnimation(anim)
    }

    override fun onResume() {
        super.onResume()
        reset()
    }

    override fun onPause() {
        super.onPause()
        stopMusic()
    }

    override fun onAnimationRepeat(p0: Animation?) {

    }

    override fun onAnimationEnd(p0: Animation?) {
        if (progress.progress == progress.max) {
            Toast.makeText(this, "OMG! You did it!", Toast.LENGTH_LONG).show()
            image.setImageResource(R.drawable.ic_explode)
            startMusic(R.raw.explosion_1)
            progress.setOnClickListener(null)

        } else {
            Toast.makeText(this, "Woohoo! You finished it!", Toast.LENGTH_LONG).show()
            image.setImageResource(R.drawable.ic_melon)
            startMusic(R.raw.wow)
        }
        Handler().postDelayed({reset()}, 10000)
    }

    override fun onAnimationStart(p0: Animation?) {

    }

    override fun onClick(p0: View?) {
        progress.progress -= 1
        if (progress.progress <= 0) {
            progress.clearAnimation()
        } else {
            anim.from = progress.progress.toFloat()
            progress.startAnimation(anim)
        }
    }
}